File: services/graphql-service/app/application/use_cases/user/delete_user_use_case.py
from uuid import UUID

from app.application.interfaces.user_repository_interface import IUserRepository

class DeleteUserUseCase:

    def __init__(
        self,
        repository: IUserRepository
    ):
        self.repository = repository

    async def execute(
        self,
        user_id: UUID
    ) -> bool:
        
        return await self.repository.delete(
            user_id
        )

File: services/grpc-service/src/GrpcService.Application/UseCases/DeleteUser/DeleteUserUseCase.cs
using GrpcService.Application.Interfaces;
using GrpcService.Domain.Exceptions;

namespace GrpcService.Application.UseCases.DeleteUser;

public sealed class DeleteUserUseCase(IUserRepository repository)
{
    public async Task ExecuteAsync(
        Guid id, 
        CancellationToken cancellationToken = default
    )
    {
        var deleted = await repository.DeleteAsync(id, cancellationToken);

        if (!deleted)
            throw new UserNotFoundException(id);
    }
}